test/unit/backend_api_logic/routing_policy_test.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module BackendApiLogic
+  class RoutingPolicyTest < ActiveSupport::TestCase
+    setup do
+      @service = FactoryBot.create(:service)
+      other_backend_api = FactoryBot.create(:backend_api, system_name: 'foo')
+      @service.backend_api_configs.create(backend_api: other_backend_api, path: 'foo')
+    end
+
+    attr_reader :service
+    delegate :proxy, to: :service
+
+    test '#with_subpaths?' do
+      assert proxy.with_subpaths?
+      other_proxy = FactoryBot.create(:proxy)
+      refute other_proxy.with_subpaths?
+    end
+
+    test '#policy_chain' do
+      injected_rules = [
+        { url: service.backend_apis.last.private_endpoint,  condition: { operations: [match: :path, op: :matches, value: '/foo/.*|/foo/?'] }, replace_path: '{{original_url.path | replace: "/foo", "/"}}' },
```

Comment:
Check this PR: https://github.com/3scale/porta/pull/1206
In this case, for `service.backend_apis.last` and the next line `service.backend_apis.first`, the order will be random for Postgres and sometimes we will get random CircleCI failures for that :disappointed: So you can fix it by something like:
```ruby
injected_rules = [
  { url: service.backend_apis.order(:id).last.private_endpoint,  condition: { operations: [match: :path, op: :matches, value: '/foo/.*|/foo/?'] }, replace_path: '{{original_url.path | replace: "/foo", "/"}}' },
  { url: service.backend_apis.order(:id).first.private_endpoint, condition: { operations: [match: :path, op: :matches, value: '/.*'] } }
]
```

Reply:
Taken care by @hallelujah already: https://github.com/3scale/porta/blob/83a66a93c9bf04a3ae383327e111c324bd1b8096/app/models/backend_api_config.rb#L6